pkg/model/openstackmodel/firewall.go

```diff
@@ -221,7 +221,7 @@ func (b *FirewallModelBuilder) addHTTPSRules(c *fi.ModelBuilderContext, sgMap ma
 	addDirectionalGroupRule(c, masterSG, nodeSG, httpsIngress)
 	addDirectionalGroupRule(c, masterSG, masterSG, httpsIngress)
 
-	if b.UseLoadBalancerForAPI() {
```

Comment:
This utility method checks whether `cluster.Spec.API.LoadBalancer` is nil.  I am unsure of the conditions in which this is true, however in create cluster it seems this is always set to an empty struct if the following conditions are true:
```
	return s.DNS == nil && s.LoadBalancer == nil
```
I am using gossip without a loadbalancer in this scenario, connecting to one of the 3 masters.  I would not consider this to be a production ready configuration which makes me question whether these changes should be warranted at all.